utxo/src/cache.rs

```diff
@@ -145,6 +152,73 @@ impl<'a> UtxosCache<'a> {
         tx_undo.map(TxUndo::new)
     }
 
+    // Marks outputs of a transaction as spent and inputs as unspent
+    pub fn disconnect_transaction(
+        &mut self,
+        tx: &Transaction,
+        tx_undo: TxUndo,
+    ) -> Result<(), Error> {
+        for (i, _) in tx.outputs().iter().enumerate() {
+            let tx_outpoint = OutPoint::new(OutPointSourceId::from(tx.get_id()), i as u32);
+            self.spend_utxo(&tx_outpoint)?;
+        }
```

Comment:
It changes utxo entries state to 'spent' in a local cache. When the cache is flushed to db, 'spent' entries are ignored and just won't get stored. I believe this can be described as spend.